src/main/scala/com/fulcrumgenomics/bam/FindSwitchbackReads.scala

```diff
@@ -0,0 +1,398 @@
+                                      maxOffset: Int,
+                                      maxErrorRate: Double): Option[ReadBasedHit] = {
+    if (maxOffset <= 0) None else template.primaryReads.filter(_.mapped).flatMap { rec =>
+        if (rec.positiveStrand && rec.cigar.head.operator == CigarOperator.S && rec.cigar.head.length >= minLength) {
+          val ref = refs(rec.refName)
+          val bases = Sequences.revcomp(rec.basesString.take(rec.cigar.head.length))
+          val windowStart = max(1, rec.start - maxOffset)
+          val windowEnd = min(ref.length(), rec.start + bases.length + maxOffset)
+          val refBases = new String(ref.getBases, windowStart - 1, windowEnd - windowStart + 1)
+          find(bases, refBases, maxErrorRate).map { hit => hit.read = Some(rec); hit.offset = -(hit.offset + windowStart - rec.start); hit }
+        }
+        else if (rec.negativeStrand && rec.cigar.last.operator == CigarOperator.S && rec.cigar.last.length >= minLength) {
+          val ref = refs(rec.refName)
+          val bases = Sequences.revcomp(rec.basesString.takeRight(rec.cigar.last.length))
+          val windowStart = max(1, rec.end - bases.length - maxOffset + 1)
+          val windowEnd = min(ref.length(), rec.end + maxOffset)
+          val refBases = new String(ref.getBases, windowStart - 1, windowEnd - windowStart + 1)
+          find(bases, refBases, maxErrorRate).map { hit => hit.read = Some(rec); hit.offset = -(rec.end - (hit.offset + windowStart + bases.length - 1)); hit }
+        }
+        else {
+          None
+        }
+      }.find(_ => true)
+  }
+
+  /**
+    * Attempts to find a switchback based on the orientation of the primary reads (FF or RR) and the distance between
+    * the end of one read (in sequencing order) and the start of the next read (in sequencing order), i.e. the gap size.
+    *
+    * @param template the template to be examined.
+    * @param maxGap the maximum allowable gap size
+    * @return either Some(TandemBasedHit) if a hit can be found, else None
+    */
+  private[bam] def findTandemSwitchback(template: Template, maxGap: Int) : Option[TandemBasedHit] =
+    if (maxGap <=0 ) None else template.pairOrientation match {
+      case Some(PairOrientation.TANDEM) =>
+        (template.r1, template.r2) match {
+          case (Some(r1), Some(r2)) =>
+            val (earlier, later) = if (r1.start <= r2.start) (r1, r2) else (r2, r1)
+            val gap = later.start - earlier.end - 1
+            if (abs(gap) <= maxGap) {
+              Some(TandemBasedHit(later.start - earlier.end - 1))
+            }
+            else {
+              None
+            }
+        }
+      case _ =>
+        None
+  }
+
+  /**
+    * Attempts to find the first ungapped match of the query sequence within the target sequence. "First" is either
+    * from the start of the target sequence, or if `preferLaterMatches` is true, from the end of the target sequence.
```

Comment:
I don't see `preferLaterMatches` used anywhere, remove it?

Reply:
To be clear, there are no calls to `find` that change the value of `preferLaterMatches` from` false.  I can only see `find` being called in `findReadBasedSwitchback`, which does not provide a value for `preferLaterMatches`